src/artery/envmod/CarIdentityRegistrant.cc

```diff
@@ -0,0 +1,41 @@
+#include "artery/envmod/CarIdentityRegistrant.h"
+#include "artery/utility/IdentityRegistry.h"
+#include "artery/traci/MobilityBase.h"
+
+
+namespace artery
+{
+
+Define_Module(CarIdentityRegistrant)
+
+void CarIdentityRegistrant::initialize()
+{
+	initializeIdentity();
+	registerIdentity();
+}
+
+void CarIdentityRegistrant::initializeIdentity()
+{
+	auto parent = this->getParentModule();
+	auto mobility = dynamic_cast<MobilityBase*>(parent->getSubmodule("mobility", -1));
```

Comment:
`mobility` may be a nullptr if no submodule named "mobility" is found or its type is not a descendant of `MobilityBase`. Throwing an exception in this case should be fine.